modules/gdscript/gdscript_compiler.cpp

```diff
@@ -2277,14 +2277,13 @@ Error GDScriptCompiler::_populate_class_members(GDScript *p_script, const GDScri
 					_set_error(vformat(R"(Could not compile base class "%s" from "%s": %s)", base->fully_qualified_name, base->path, error_names[err]), nullptr);
 					return err;
 				}
-				if (base_root.is_valid()) {
+				if (!base_root.is_null()) {
```

Comment:
This changes nothing, before/after are equivalent.
https://github.com/godotengine/godot/blob/860884bde7d895887236306887d01381f747bcd7/core/object/ref_counted.h#L212-L213
And I think `is_valid()` is preferred in here, that's kinda why this method exists (to avoid having negated `is_null()` checks).

Reply:
You're right for `Ref<T>` instances, except for `GDScript`.
> https://github.com/godotengine/godot/blob/860884bde7d895887236306887d01381f747bcd7/modules/gdscript/gdscript.h#L185

Reply:
but thats defned in `Script`, not `Ref` https://github.com/godotengine/godot/blob/860884bde7d895887236306887d01381f747bcd7/core/object/script_language.h#L144

Reply:
You're right. My bad.

Reply:
`base_root` is `Ref<GDScript>` so:
- `base_root.is_valid()` calls `Ref::is_valid`,
- `base_root->is_valid()` calls `GDScript::is_valid`.

Reply:
> `base_root` is `Ref<GDScript>` so:
> 
>     * `base_root.is_valid()` calls `Ref::is_valid`,
> 
>     * `base_root->is_valid()` calls `GDScript::is_valid` (or its override as it's virtual).

I got mixed up. Thanks.

Reply:
i don't blame you its **very** confusing

Reply:
Wow this is an API asking for bugs...